app/lib/audit/backend.dart

```diff
@@ -136,4 +139,65 @@ class AuditBackend {
     InvalidInputException.check(parsed != null, 'Unable to parse `before`.');
     return parsed!;
   }
+
+  /// Returns the entries from the last day.
+  ///
+  /// Keeps the [_cachedRecords] fields updated, and lists only the entries
+  /// up to the last query.
+  ///
+  /// NOTE: there is no guarantee that the entries are in creation order
+  Future<List<AuditLogRecord>> getEntriesFromLastDay() async {
+    if (_cacheRecordsUpdateFuture != null) {
+      await _cacheRecordsUpdateFuture;
+    } else {
+      final now = clock.now().toUtc();
+      final cachedAge = now.difference(_cachedRecords.updated);
+      if (cachedAge.inSeconds < 3) {
+        return _cachedRecords.records;
+      }
+
+      // we should have only one update running
+      _cacheRecordsUpdateFuture = _updateEntriesFromLastDay(
+        cachedAge: cachedAge,
+        oldRecords: _cachedRecords.records,
+      );
+      await _cacheRecordsUpdateFuture;
+      _cacheRecordsUpdateFuture = null;
+    }
+    return _cachedRecords.records;
+  }
+
+  Future<void> _updateEntriesFromLastDay({
+    required Duration cachedAge,
+    required Iterable<AuditLogRecord> oldRecords,
+  }) async {
+    // calculate window to query
+    final now = clock.now();
+    final day = const Duration(days: 1);
+    var window = cachedAge > day ? day : (day - cachedAge);
+    if (window < Duration(minutes: 2)) {
+      window = Duration(minutes: 2);
+    }
+
+    final query = dbService.query<AuditLogRecord>()
+      ..filter('created >', now.subtract(window));
+    final current = await query.run().toList();
+
+    // merge records from cache and current query
+    final currentIds = current.map((e) => e.id!).toSet();
+    final records = [
+      ...oldRecords.where((r) =>
+          !currentIds.contains(r.id!) && now.difference(r.created!) < day),
```

Comment:
nit, not sure if it's formatted any better :D
```suggestion
      ...oldRecords
          .where((r) => !currentIds.contains(r.id!)),
          .where((r) => now.difference(r.created!) < day),
```